src/qmapshack/gis/wpt/CGisItemWpt.cpp

```diff
@@ -285,6 +286,29 @@ void CGisItemWpt::newWpt(const QPointF& pt, const QString &name, const QString &
         wpt->setDescription(desc);
     }
 
+    cfg.setValue("Waypoint/lastName", wpt->getName());
+    cfg.setValue("Waypoint/lastIcon", wpt->getIconName());
+}
+
+void CGisItemWpt::newWpt(const poi_t &poi, IGisProject *project, bool openEditWIndow)
+{
+    SETTINGS;
+    const QString& _icon = poi.icon.isEmpty() ? cfg.value("Waypoint/lastIcon", "Waypoint").toString() : poi.icon;
+    const QString& _name = poi.name.isEmpty() ? getLastName("") : poi.name;
+
+    CGisItemWpt * wpt = new CGisItemWpt(poi.pos * RAD_TO_DEG, _name, _icon, project);
```

Comment:
I would suggest to set the new waypoint initialy to read-only. By that the information in the waypoint's edit dialog is more compact. And most likely no one will edit the waypoint anyways.

`wpt->setReadOnlyMode(true);`